pkg/agent/datamodel/sig_config.go

```diff
@@ -12,6 +16,151 @@ const (
 	AzurePublicCloudSigSubscription string = "109a5e88-712a-48ae-9078-9ca8b3c81345" // AKS VHD
 )
 
+//nolint:gochecknoglobals
+var (
+	CachedFromComponentContainerImages = make(map[string]ContainerImage)
+	CachedFromComponentDownloadedFiles = make(map[string]DownloadFiles)
+	CachedFromManifest                 = make(map[string]ProcessedManifest)
+)
+
+//nolint:gochecknoinits
+func init() {
+	manifest := CacheManifest()
+	processManifest(manifest)
+
+	components := CacheComponents()
+	processComponents(components)
+}
+
+func CacheManifest() Manifest {
+	_, filename, _, _ := runtime.Caller(0)
+	manifestFilePath := "../../../parts/linux/cloud-init/artifacts/manifest.json"
+	return getCachedVersionsFromManifestJSON(path.Join(path.Dir(filename), manifestFilePath))
+}
+
+func CacheComponents() Components {
+	_, filename, _, _ := runtime.Caller(0)
+	componentsFilePath := "../../../vhdbuilder/packer/components.json"
+	return getCachedVersionsFromComponentsJSON(path.Join(path.Dir(filename), componentsFilePath))
+}
+
+func processManifest(manifest Manifest) {
+	CachedFromManifest["kubernetes"] = ProcessedManifest{
+		Versions: manifest.Kubernetes.Versions,
+	}
+	CachedFromManifest["runc"] = ProcessedManifest{
+		Versions:  manifest.Runc.Versions,
+		Pinned:    manifest.Runc.Pinned,
+		Installed: manifest.Runc.Installed,
+	}
+	CachedFromManifest["containerd"] = ProcessedManifest{
+		Versions: manifest.Containerd.Versions,
+		Pinned:   manifest.Containerd.Pinned,
+		Edge:     manifest.Containerd.Edge,
+	}
+	CachedFromManifest["nvidia-container-runtime"] = ProcessedManifest{
+		Versions: manifest.NvidiaContainerRuntime.Versions,
+	}
+	CachedFromManifest["nvidia-drivers"] = ProcessedManifest{
+		Versions: manifest.NvidiaDrivers.Versions,
+	}
+}
+
+func processComponents(components Components) {
+	for _, image := range components.ContainerImages {
+		componentName, err := processContainerImage(image.DownloadURL)
+		if err != nil {
+			panic(err)
+		}
+		CachedFromComponentContainerImages[componentName] = image
+	}
+	for _, file := range components.DownloadFiles {
+		componetName, err := processDownloadFile(file.DownloadURL)
+		if err != nil {
+			panic(err)
+		}
+		CachedFromComponentDownloadedFiles[componetName] = file
+	}
+}
+
+func getCachedVersionsFromManifestJSON(manifestFilePath string) Manifest {
+	data, err := os.ReadFile(manifestFilePath)
+	if err != nil {
+		panic(err)
+	}
+	data = bytes.ReplaceAll(data, []byte("#EOF"), []byte(""))
+	var manifest Manifest
+	if err = json.Unmarshal(data, &manifest); err != nil {
+		panic(err)
+	}
+	return manifest
+}
+
+func getCachedVersionsFromComponentsJSON(componentsFilePath string) Components {
+	data, err := os.ReadFile(componentsFilePath)
+	if err != nil {
+		panic(err)
+	}
+	var components Components
+	if err = json.Unmarshal(data, &components); err != nil {
+		panic(err)
+	}
+	return components
+}
+
+type Manifest struct {
```

Comment:
this type along with `ProcessedManifest` feels redundant

why not just do 

```
type Dependency struct {
	Versions []string `json:"versions"`
         Installed map[string]string `json:"installed"`
         Pinned map[string]string `json:"pinned'`
         Edge string `json:"edge"`
}

type Manifest struct {
       Containerd Dependency `json:"containerd"`
       Runc Dependency `json:"runc"`
       NvidiaContainerRuntime `json:"nvidia-container-runtime"`
       NvidiaDrivers `json:"nvidia-drivers"`
       Kubernetes `json:"kubernetes"`
}
```

Reply:
https://github.com/Azure/AgentBaker/blob/master/schemas/manifest.cue

Reply:
it also make me sad since I'm trying to rework that cue schema for components/manifests to support the effort for configurable private builds; if you're gonna be picking up some of the security work then I might end up handing that off to you :)

for ref: https://github.com/Azure/AgentBaker/pull/4171